core/src/test/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverActionTests.java

```diff
@@ -0,0 +1,192 @@
+        String sourceAlias = randomAsciiOfLength(10);
+        String optionalTargetAlias = randomAsciiOfLength(10);
+        String sourceIndex = randomAsciiOfLength(10);
+        String targetIndex = randomAsciiOfLength(10);
+        final RolloverRequest rolloverRequest = new RolloverRequest(sourceAlias, optionalTargetAlias);
+        final IndicesAliasesClusterStateUpdateRequest updateRequest =
+            TransportRolloverAction.prepareIndicesAliasesRequest(sourceIndex, targetIndex, rolloverRequest);
+
+        final AliasAction[] actions = updateRequest.actions();
+        assertThat(actions.length, equalTo(3));
+        boolean foundAdd = false;
+        boolean foundRemove = false;
+        for (AliasAction action : actions) {
+            if (action.actionType() == AliasAction.Type.ADD) {
+                foundAdd = true;
+                assertThat(action.index(), equalTo(targetIndex));
+                assertThat(action.alias(), anyOf(equalTo(sourceAlias),
+                    equalTo(optionalTargetAlias)));
+            } else if (action.actionType() == AliasAction.Type.REMOVE) {
+                foundRemove = true;
+                assertThat(action.index(), equalTo(sourceIndex));
+                assertThat(action.alias(), equalTo(sourceAlias));
+            }
+        }
+        assertTrue(foundAdd);
+        assertTrue(foundRemove);
+    }
+
+    public void testValidation() throws Exception {
+        String index1 = randomAsciiOfLength(10);
+        String alias = randomAsciiOfLength(10);
+        String index2 = randomAsciiOfLength(10);
+        String aliasWithMultipleIndices = randomAsciiOfLength(10);
+        final Settings settings = Settings.builder()
+            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+            .put(IndexMetaData.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
+            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
+            .build();
+        final MetaData metaData = MetaData.builder()
+            .put(IndexMetaData.builder(index1)
+                .settings(settings)
+                .putAlias(AliasMetaData.builder(alias))
+                .putAlias(AliasMetaData.builder(aliasWithMultipleIndices))
+            )
+            .put(IndexMetaData.builder(index2)
+                .settings(settings)
+                .putAlias(AliasMetaData.builder(aliasWithMultipleIndices))
+            ).build();
+
+        try {
+            TransportRolloverAction.validate(metaData, new RolloverRequest(aliasWithMultipleIndices,
+                randomBoolean() ? null : randomAsciiOfLength(10)));
+            fail("expected to throw exception");
+        } catch (IllegalArgumentException e) {
+            assertThat(e.getMessage(), equalTo("source alias maps to multiple indices"));
+        }
+
+        try {
+            TransportRolloverAction.validate(metaData, new RolloverRequest(randomFrom(index1, index2),
```

Comment:
Probably should use `expectThrows`. Is marginally easier to read and less typing to maintain.